### PR TITLE
Support line grouping and editing for line-level annotations

### PR DESCRIPTION
**Pull request by blms**
see also https://github.com/Princeton-CDH/geniza/pull/1537

## In this PR
- Support for line and block level annotation editing in new "line mode"

## Notes
- The biggest change here is actually a new custom element that's _just_ for editing an annotation label. That's because block-level annotations must not be otherwise editable in line mode. (Conversely, individual line-level annotations have no labels themselves and thus the label editor is disabled for them.) 
  - It ends up having many methods of `AnnotationBlock` written in a similar, but not identical, way—unfortunately I ultimately couldn't easily leverage polymorphism.
- In line mode, I disabled most functionality except for editing text contents of line-level annotations, and label content of block-level annotations—mostly to avoid dealing with complications and bugs that arise from the rest of the functionality. I figure we can add them back in one by one if/when needed in later passes.
- There were some strange interactions between blocks and lines contained within, namely that the two were constantly z-fighting and it seemed random which would be selected when you clicked. So I decided just to not display the block-level annotations in Annotorious when in line mode, though their geometry is being stored.
- Note that the cropped line image by each line of text from your original proof of concept is not included at this point.
- Thoughts on whether this needs unit testing? My inclination is not to write them, since this may be somewhat provisional.

### CODE REVIEW

File: src/elements/AnnotationBlock.ts [hunk 1]
```diff
@@ -88,7 +88,10 @@ class AnnotationBlock extends HTMLElement {
             this.labelElement.innerHTML = this.annotation.body[0].label || "";
             this.bodyElement.innerHTML = this.annotation.body[0].value;
         }
-        this.append(this.labelElement);
+        if (this.annotation.textGranularity !== "line") {
```
Comment (rlskoeser) [inline src/elements/AnnotationBlock.ts:91]: would it make sense to define a constant for supported text granularity values?

File: src/index.ts [hunk 2]
```diff
@@ -70,6 +80,8 @@ class TranscriptionEditor {
         this.storage = storage;
         // disable the default annotorious editor (headless mode)
         this.anno.disableEditor = true;
+        // read only in line mode
+        this.anno.readOnly = this.storage.settings.lineMode;
```
Comment (rlskoeser) [inline src/index.ts:84]: nice

File: src/index.ts [hunk 3]
```diff
@@ -124,11 +139,14 @@ class TranscriptionEditor {
             Editor();
         }
         if (!window.tinyConfig) {
+            // hide numberded list in line-level editor mode
```
Comment (rlskoeser) [inline src/index.ts:142]: tiny typo in comment

```suggestion
            // hide numbered list in line-level editor mode
```

**Review by rlskoeser — APPROVED** [review]
Wow, thank you for all your work moving this forward - complicated stuff, but exciting.

From looking at the code I'm having a little trouble understanding the difference between an `AnnotationLabel` and an `AnnotationBlock` - there's more context in your PR notes but I want to be sure I'm understanding (and maybe we can slightly expand the comments). Is the new label object is so you can edit the label associated with a group of line-level annotations separately, since the block itself is not editable, only the lines?

I like the rename to the more general `annotationElement`, just want to be sure I understand (and we document) what the differences are between these two cases.

Fine with me to hold off on unit testing for now; and even the duplication of code also seems totally reasonable at this stage of implementation.

I've only reviewed on github, let me know if there's anything you want me to pull and test locally.

**Comment by blms** [comment]
> Is the new label object is so you can edit the label associated with a group of line-level annotations separately, since the block itself is not editable, only the lines?

Exactly right, thanks for describing this succinctly and clearly!

**Pull request merged**